daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -401,6 +388,12 @@ class DataProcessor private[japi] (private var dp: SDataProcessor)
   extends WithDiagnostics(dp)
   with Serializable {
 
+  /**
+   * For use by the JAPI implementation only.
+   * @return the underlying data processor
+   */
+  private [japi] def getUnderlyingDataProcessor = dp
```

Comment:
Does scala's package private thing apply to the generated java docs? If not, we might want to add a doc that says this shouldn't be used externally.